ict-soil-dendrometer-probe/transformers/main/source.js

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+ * 
+ *  Transformation code for ICT Dendrometer & Soil Moisture Probe 
+ * 
+ * ****************************************************************************/
+const STEM_DIAMETER = 0;
+const STEM_TEMPERATURE = 1;
+const VWC_COUNT = 2;
+const TEMPERATURE = 3;
+const EC = 4;
+const VWC = 5;
+
+/**
+ * Extracts bytes from byte_payload from start to end and converts these bytes 
+ * into a float. Adapted from: 
+ * https://www.thethingsnetwork.org/forum/t/decode-float-sent-by-lopy-as-node/8757/2
+ */
+function extractFloat(byte_payload, start, end) {
+  // Extract bytes from payload, converting each byte into its unicode value
+  var bytes = [];
+  for(var i = start; i < end; i++) 
+    bytes.push(byte_payload[i].charCodeAt(0));
+
+  // Convert bytes to a float
+  var bits = bytes[0]<<24 | bytes[1]<<16 | bytes[2]<<8 | bytes[3];
+  var sign = (bits>>>31 === 0) ? 1.0 : -1.0;
+  var e = bits>>>23 & 0xff;
+  var m = (e === 0) ? (bits & 0x7fffff)<<1 : (bits & 0x7fffff) | 0x800000;
+  var float = sign * m * Math.pow(2, e - 150);
+
+  // Return float unrounded
+  return float*100/100;
```

Comment:
This rounding is still there.

Reply:
To clarify, is this in reference to the return statement? Functionally, it should still be returning the same thing with or without the *100/100. I have removed that part regardless so that it is just returning float.

Reply:
All good, there is a loss of precision by doing return *100/100